modules/setting/setting.go

```diff
@@ -1159,6 +1159,19 @@ func CreateOrAppendToCustomConf(callback func(cfg *ini.File)) {
 	if err := cfg.SaveTo(CustomConf); err != nil {
 		log.Fatal("error saving to custom config: %v", err)
 	}
+
+	// Change permissions to be more restrictive
+	fi, err := os.Stat(CustomConf)
+	if err != nil {
+		log.Error("Failed to determine current conf file permissions: %v", err)
+		return
+	}
+
+	if fi.Mode().Perm() > 0o600 {
+		if err = os.Chmod(CustomConf, 0o600); err != nil {
```

Comment:
Whilst I'd agree that we should create with mode 0600 and perhaps make a non-failing advisory doctor alert - going around changing the mode of files seems a bit rude.

Certainly we'd need to check that gitea is the owner of the file before attempting to change the mode - I could see a configuration that has gitea not being the owner but being able to read it through group permissions.

Reply:
Its non failing and just log with warn - but the user check is a good point!

Reply:
> and perhaps make a non-failing advisory doctor alert

What do you mean exactly? The cli command?

> I could see a configuration that has gitea not being the owner but being able to read it through group permissions.

I see... changing from currently `-rw-r--r--` to `-rw-rw----`. This would make it both better and worse in my opinion. Removing read permissions from "others" is great. Give write permissions to a group should be considered carefully. Especially without any containerized context, multiple users might have the same group permissions and would be able to change the file content.
Gitea supports setting the user to run with/under but not the group, if I'm not wrong.

> going around changing the mode of files seems a bit rude

Got your point with that. Do you think the permission modification should only be done when the file is created and not possibly each time the content changes? I'm not sure, if this method is called in both cases. Probably...
Would it be ok to add a check if the file was created right before this permission change?

Reply:
I'm not suggesting we change to 0660 - gods no - I'm just saying that there may be configurations out there that rely on it. 

---

I think if you just add a check to ensure that Gitea owns the file - and Warns if not then we're fine as this is only in the rewrite section anyway.